vars/checkResultsPublish.groovy

```diff
@@ -0,0 +1,147 @@
+import com.cloudbees.groovy.cps.NonCPS
+import com.sap.piper.Utils
+
+/**
+ * checkResultsPublish
```

Comment:
I would suggest to rename it to `checksPublishResults` since we aimed for a naming like toolTaskDetails. 

Sometimes we don't have the details since we have rather generic steps like `mavenExecute` or `dockerExecute` but one could think of something like `mavenExecuteBuild`, ... further down the road